filestore/handlers.py

```diff
@@ -38,7 +38,11 @@ def __call__(self, point_number):
                                           file_number)
                 spe_obj = PrincetonSPEFile(fname)
                 self._f_cache[file_number] = spe_obj
-            out_stack.append(self._f_cache[file_number][frame_no])
+            spe = self._f_cache[file_number]
+            if len(spe) > 1:
```

Comment:
I that case the whole loop structure should go away and the resource does
not need to know frames per point.

On Sat, Mar 7, 2015, 07:25 Stuart Wilkins <notifications@github.com> wrote:

> In filestore/handlers.py
> <https://github.com/tacaswell/filestore/pull/1#discussion_r25995972>:
>
> > @@ -38,7 +38,11 @@ def __call__(self, point_number):
> >                                            file_number)
> >                  spe_obj = PrincetonSPEFile(fname)
> >                  self._f_cache[file_number] = spe_obj
> > -            out_stack.append(self._f_cache[file_number][frame_no])
> > +            spe = self._f_cache[file_number]
> > +            if len(spe) > 1:
>
> No, we get one SPE file per datum. If you set (I was just confirming this)
> the number of images per datum (data point) to 10 then you get 10x2048x2048
> images. So we don't have to do any crazy counting, but we should return the
> whole stack and not truncate to one image.
>
> I tested this and it works OK, however there may be a better way to do
> this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tacaswell/filestore/pull/1/files#r25995972>.
>